cpp/src/arrow/extension_type.cc

```diff
@@ -145,12 +146,17 @@ static void CreateGlobalRegistry() {
   g_registry = std::make_shared<ExtensionTypeRegistryImpl>();
 
 #ifdef ARROW_JSON
```

Comment:
I think the `ARROW_JSON` was only required because we were having a bug on CMake that you are fixing right now. Why should `fixed_shape_tensor` has nothing to do with JSON?

Reply:
`fixed_shape_tensor` uses rapidjson to serialize and deserialize metadata (see `FixedShapeTensorType::Serialize` and `FixedShapeTensorType::Deserialize`) of the type so we need `ARROW_JSON` to build it. UUID has no metadata to serialize so we don't require JSON. It's annoying that for some builds we get all extensions and for some only a subset, but I don't have a better idea.